src/Network/Network/ApplicationGateway/GetAzureApplicationGatewayAvailableRequestHeaders.cs

```diff
@@ -0,0 +1,39 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using AutoMapper;
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.WindowsAzure.Commands.Common;
+using System.IO;
+using System.Management.Automation;
+using System.Reflection;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApplicationGatewayAvailableRequestHeaders"), OutputType(typeof(string))]
```

Comment:
@nirkum18 we cannot return a `string` object from cmdlets as this will affect piping scenarios. We should create new types for each property that we can return instead that contains properties that the user would find useful.

Reply:
@cormacpayne the server returns an array of strings for all of those new cmdlets (e.g. `[ "A-IM", "Accept", "Accept-Charset"]`), how should we proceed here? I don't think there's a need for custom types here

Reply:
@nirkum18 @number213 the reason for advising against `string` output types is because it can cause issues with parameter binding when used in piping scenarios -- it is much easier in PowerShell to use concrete objects and pass those around then using strings. As mentioned in the design review, the ideal pattern would be to bundle these cmdlets into one, introducing a new concrete type that has the result of these cmdlets as properties of the type, and users would only have to call one cmdlet to get an object back that has all of these values. It's mentioned that these cmdlets are following the pattern of existing cmdlets, but those cmdlets return concrete types from their API calls, whereas these calls all return a list of strings, so it would be beneficial to bundle them together for a better user experience.

Reply:
Sounds reasonable to me. @nirkum18, this is your feature, so please consider Cormac's feedback

Reply:
@nirkum18 any update on this?

Reply:
@cormacpayne The same cmdlet behavior is implemented in the service side, swagger and powershell review doc. So we would like to keep the cmdlets consistent at all places.